### PR TITLE
Bump SDK versions

### PR DESCRIPTION
**Pull request by jonathan-caryl**
And extract more version numbers in to the project root, to make them
easier to update in the future

### CODE REVIEW

File: remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/widget/number/FloatSeekBar.java [hunk 1]
```diff
@@ -27,7 +28,7 @@
  * are all zero-based integers. We need a more versatile solution that supports float. To do this we
  * build on top of the existing seek bar, allowing us to set a minimum, maximum, and step sizes.
  */
-public class FloatSeekBar extends SeekBar {
+public class FloatSeekBar extends AppCompatSeekBar {
```
Comment (jonathan-caryl) [inline remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/widget/number/FloatSeekBar.java:31]: We now get a lint failure if we extend from a regular `SeekBar`

File: remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/widget/ColorListVariableWidget.java [hunk 2]
```diff
@@ -97,6 +97,10 @@ void setColorItem(final ColorItem colorItem) {
   @VisibleForTesting
   static class Adapter extends RecyclerView.Adapter<ViewHolder> {
 
+    Adapter() {
```
Comment (jonathan-caryl) [inline remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/widget/ColorListVariableWidget.java:100]: workaround for overzealous lint

**Comment by miguelandres** [comment]
This breaks unit tests, can you debug the issue locally and update your pr?

**Comment by jonathan-caryl** [comment]
@miguelandres I've tweaked this to not update the gradle version, as that seems to be what breaks the tests. I've found various 'fixes' online, none of which seemed to work in this case, so I figure it's best to change the gradle version in another PR when I've figured out exactly what's going on.

File: build.gradle [hunk 3]
```diff
@@ -27,7 +27,6 @@ buildscript {
     classpath 'com.android.tools.build:gradle:2.2.3'
     classpath 'com.github.dcendents:android-maven-gradle-plugin:1.4.1'
     classpath 'com.google.gms:google-services:3.0.0'
-    classpath 'com.neenbedankt.gradle.plugins:android-apt:1.8'
```
Comment (jonathan-caryl) [inline build.gradle:30]: I was going to replace this, as we can now just use `annotationProcessor` but actually it wasn't being used at all.

**Pull request merged**